lib/foodcritic/linter.rb

```diff
@@ -261,13 +261,30 @@ def cookbook_dir(file)
       cook_val = @dir_cache[abs_file]
       return cook_val unless cook_val.nil?
 
-      # we didn't find something in cache so look it up and cache it for later
-      cook_val = Pathname.new(File.join(File.dirname(file),
-                                        case File.basename(file)
-                                        when "metadata.rb" then ""
-                                        when /\.erb$/ then "../.."
-                                        else ".."
-                                        end)).cleanpath
+      if file =~ /\.erb$/
+        # split each directory into an item in the array
+        dir_array = File.dirname(file).split(File::SEPARATOR)
+
+        # walk through the array of directories backwards until we hit the templates directory
+        position = -1
+        position -= 1 until dir_array[position] == "templates"
+
+        # go back 1 more position to get to the cookbook dir
+        position -= 1
```

Comment:
This and 269 should/could be the same no?

Reply:
if we're at say -3 already we need to make sure we're at -4. -3 would be the templates directory